### PR TITLE
Actually implemented surveys

### LINKED ISSUE #47: Support survey / results
# Goal
Implement the correct survey and a way for admins to see the results

# Description
We need to change the existing survey endpoint from just taking a string to actually parsing it, and store the data in the database. The fields are:

Question | Type | Return format
--- | --- | ---
How many times would you visit this exhibit? | `number` | Bar chart, one for each number (percent fo respondents who picked that number)
Who do you think would enjoy this exhibit? (male, female, kids, teenagers, adults) | `{ population: boolean }` | Bar chart for each population (percent of respondents who picked that population)
How likely are you to recommend this exhibit to a friend? | `number` [1-5] | [NPS dial](https://i.imgur.com/nxXerUM.png)

# Additional notes
No implementation hints this time, it's pretty straightforward.

### PR DESCRIPTION
**Pull request by nic-hartley**
Fixes #47

### CODE REVIEW

File: src/main/java/org/comic_conmuseum/fan_forge/backend/persistence/SupportQueryBean.java [hunk 1]
```diff
@@ -34,8 +40,9 @@ public void setupTable(boolean reset) {
                 "    sid SERIAL PRIMARY KEY, " +
                 "    exhibit SERIAL REFERENCES exhibits(eid) ON DELETE CASCADE ON UPDATE CASCADE, " +
                 "    supporter TEXT ,"+//TODO SERIAL REFERENCES users(uid) ON DELETE CASCADE ON UPDATE CASCADE, " +
-                // TODO Get actual survey data fields to use and use them
-                "    survey_data TEXT, " +
+                "    visits INTEGER NOT NULL CHECK (0 <= visits AND visits <= 10), " +
+                "    rating INTEGER NOT NULL CHECK (0 <= rating AND rating <= 10), " +
+                "    " + POPULATIONS_COLUMN_DEFS + ", " +
```
Comment (zachtjones) [inline src/main/java/org/comic_conmuseum/fan_forge/backend/persistence/SupportQueryBean.java]: From the API docs:
visits: integer // [1-10] How many times the respondent would visit this exhibit
rating: integer // [1-10] Rating given by the user, for NPS calculation

I think you should change these to be 0 < visits, or 1 <= visits. Same with rating

File: src/main/java/org/comic_conmuseum/fan_forge/backend/models/Survey.java [hunk 2]
```diff
@@ -2,13 +2,35 @@
...
+import java.util.HashMap;
+import java.util.Map;
 
 public class Survey {
+    // These should _never_ be referenced directly if at all possible. It
+    // must also never, ever be edited. If you edit it, I will slap you.
+    public static final String[] POPULATIONS = {
+            "male", "female", "kids", "teenagers", "adults"
+    };
+
```
Comment (zachtjones) [inline src/main/java/org/comic_conmuseum/fan_forge/backend/models/Survey.java]: If you make `POPULATIONS` an enum, then you can 
 - still do the .values() to get all of them
 - create getter methods for:
    - how it's stored in the DB ("pop_" + pop + " BOOLEAN NOT NULL")
    - field name in the DB ("pop_" + pop)
    - the map for parameters in SQL (":pop_" + pop)
 - You could also create static methods on the enum that do the .map(...).collect(Join thing) to keep the code organized better

Comment (nic-hartley) [reply src/main/java/org/comic_conmuseum/fan_forge/backend/models/Survey.java]: ooh I knew there was a better way, thanks!

File: src/main/java/org/comic_conmuseum/fan_forge/backend/persistence/SupportQueryBean.java (hunk 1 above)
Comment (nic-hartley) [reply src/main/java/org/comic_conmuseum/fan_forge/backend/persistence/SupportQueryBean.java]: Whoops, I think I made the mistake in the API docs there, actually. I'll fix that.

**Pull request merged**